pandas/tests/io/test_sql.py

```diff
@@ -1373,6 +1373,67 @@ def insert_on_conflict(table, conn, keys, data_iter):
         pandasSQL.drop_table("test_insert_conflict")
 
 
+@pytest.mark.parametrize("conn", all_connectable)
+def test_to_sql_on_public_schema(conn, request):
+    if "sqlite" in conn:
+        request.node.add_marker(
+            pytest.mark.xfail(
+                reason="sqlite database does not support schemas",
+            )
+        )
+
+    conn = request.getfixturevalue(conn)
+
+    from sqlalchemy.engine import Engine
+    from sqlalchemy.sql import text
+
+    if "mysql" in conn:
+        create_sql = text(
+            """
+        CREATE TABLE test_public_schema (
+            a  integer PRIMARY KEY,
+            b  numeric,
+            c  text
+        );
+        """
+        )
+    else:
+        create_sql = text(
+            """
+        CREATE DATABASE IF NOT EXISTS public;
+        CREATE TABLE public.test_public_schema (
+            a  integer PRIMARY KEY,
+            b  numeric,
+            c  text
+        );
+        """
+        )
+
+    if isinstance(conn, Engine):
+        with conn.connect() as con:
+            with con.begin():
+                con.execute(create_sql)
+    else:
+        with conn.begin():
+            conn.execute(create_sql)
+
+    test_data = DataFrame([[1, 2.1, "a"], [2, 3.1, "b"]], columns=list("abc"))
+    test_data.to_sql(
+        name="test_public_schema",
+        con=conn,
+        if_exists="append",
+        index=False,
+        schema="public",
+    )
+
+    df_out = sql.read_sql_table("test_public_schema", conn, schema="public")
+    assert test_data.equals(df_out)
+
+    # Cleanup
```

Comment:
The fixture is supposed to take care of the cleanup for you - are you not seeing that?

Reply:
No I wasn't seeing that, unfortunately I have no experience with pytest. Took this issue as a learning experience and did get to learn a lot. Found out how its working. Removed this cleanup part but do look at my other reply please.

Reply:
Cool no worries. I just wanted to double check as a reviewer - its hard to see the issues that lead developers down certains paths at times.

> Took this issue as a learning experience and did get to learn a lot.

That's great and we appreciate the contribution